project/api/models.py

```diff
@@ -182,3 +183,55 @@ def clean(self, *args, **kwargs):  # pylint: disable=W0221
     def save(self, *args, **kwargs):  # pylint: disable=W0221
         self.full_clean()
         super(SpacetimeVolume, self).save(*args, **kwargs)
+
+class Narrative(models.Model):
+    """
+    Stores narrative information.
+    """
+
+    author = models.CharField(max_length=100)
+    title = models.CharField(max_length=200)
+    description = models.TextField()
+    start_date = models.CharField(max_length=100)
+    end_date = models.CharField(max_length=100)
+    language = models.CharField(max_length=100)
+    tags = ArrayField(models.CharField(max_length=100)) 
+
+
+    def save(self, *args, **kwargs):  # pylint: disable=W0221
+        self.full_clean()
+        super(Narrative, self).save(*args, **kwargs)
+
+class MapSettings(models.Model):
+    """
+    Stores settings to be used when a narration is active.
+    """
+
+    bbox = models.MultiPointField()
```

Comment:
I don't think it's possible to store a tuple of tuples in a django model, I would probably use a `PolygonField` here with a mix of the [extent](https://docs.djangoproject.com/en/2.1/ref/contrib/gis/geoquerysets/#extent) property and `from_bbox` in the API for inputs and outputs in that format.

Reply:
This sounds good. We do need to add a constraint on the Polygon to be a rectangle.

Actually `MultiPointField()` as proposed by Ata might be enough?

Reply:
Yeah I think so actually. Just add a constraint to enforce exactly 4 points.

Reply:
I'd say 2 points, to follow the `xmin,ymin,xmax,ymax` thing. So SW and NE points are enough to define a bbox.

Reply:
Ah, you're right.

Reply:
Hmm, maybe we can still store the bbox in settings but it would mean the area of interest. The front end could find the best way to fit that bbox to the browser in devices with different resolution?

Reply:
I'll do my best, no worries.
For current two hand-made narratives bbox would be fine enough.